<?xml version="1.0"?>
<launch>
  <!-- <arg name="uri" default="radio://0/80/2M" /> -->
  <arg name="uri" default="usb://0" />
  <arg name="joy_dev" default="/dev/input/js0"/>
  <arg name="frame" default="/mocap" />

  <include file="$(find crazyflie_driver)/launch/crazyflie_server.launch">
  </include>

  <group ns="crazyflie">
    <node pkg="crazyflie_driver" type="crazyflie_add" name="crazyflie_add" output="screen">
      <param name="uri" value="$(arg uri)" />
      <param name="tf_prefix" value="crazyflie" />
      <param name="enable_logging" value="True"/>
      <param name="enable_logging_battery" value="False" />
      <param name="enable_logging_imu" value="True"/>
      <param name="enable_logging_magnetic_field" value="False" />
      <param name="enable_logging_packets" value="False" />
      <param name="enable_logging_pose" value="False" />
      <param name="enable_logging_pressure" value="False" />
      <param name="enable_logging_temperature" value="False" />
      <rosparam>
        genericLogTopics: ["log1"]
        genericLogTopicFrequencies: [100]
        genericLogTopic_log1_Variables: ["stateEstimate.x", "stateEstimate.y", "stateEstimate.z"]
      </rosparam>
    </node>    

    <node name="joy" pkg="joy" type="joy_node" output="screen" >
      <param name="dev" value="$(arg joy_dev)" />
      <param name="deadzone" value="0.05" />
    </node>

    <include file="$(find crazyflie_demo)/launch/logitech.launch">
    </include>

    <node name="crazyflie_demo_controller" pkg="crazyflie_demo" type="controller.py" output="screen">
        <param name="use_crazyflie_controller" value="True" />
    </node>

    <node name="pose" pkg="crazyflie_demo" type="publish_external_position_eraptor.py" output="screen">
    </node>

    <node pkg="tf" type="static_transform_publisher" name="baselink_broadcaster" args="0 0 0 0 0 0 1 $(arg frame) /crazyflie/base_link 100" />
  </group>

  <!-- run eraptor bridge -->
  <include file="$(find cortex_stream)/launch/cortex_stream.launch"/>

  <!-- run rviz -->
  <!-- <param name="robot_description" command="$(find xacro)/xacro.py $(find crazyflie_description)/urdf/crazyflie.urdf.xacro" /> -->
  <!-- <node name="rviz" pkg="rviz" type="rviz" args="-d $(find crazyflie_demo)/launch/crazyflie_pos.rviz" required="true" /> -->

  <!-- <node pkg="rqt_plot" type="rqt_plot" name="rqt_plot1" args="/crazyflie/log1/values[0] /crazyflie/log1/values[1] /crazyflie/log1/values[2]"/> -->

</launch>
